### ISSUE TITLE
Cost Optimization

### ISSUE THREAD

**Issue by lenisha** [microsoft/AzureTRE#945]
**Is your feature request related to a problem? Please describe.**
Currently cost associated with core management services is pretty high for research teams, need a way to optimize.

**Describe the solution you'd like**
Be able to provision lower SKU for some services like Service Bus
Be able to start/stop services when environment not in use - e.g Firewall, App Service etc

- [x] #972
- [ ] #2723

**Comment by marrobi** [comment]
@lenisha thanks.

I'm not sure we can start/stop firewall and/or app service plans - the functionality doesn't exist - they incur costs whenever deployed. 

We could reduce the sku on some services - such as app service and service bus - as a compromise with security. But something like firewall is harder as we can't stop it, and if we delete it we lose all of the existing rules.

What do you think is a good approach?

**Comment by lenisha** [comment]
@marrobi 
the thought was around utilizing start/stop to save cost wherever possible, for example we use it for Firewall and significantly drops price tag:
https://docs.microsoft.com/en-us/azure/firewall/firewall-faq#how-can-i-stop-and-start-azure-firewall

and same with app gateway:
https://docs.microsoft.com/en-us/powershell/module/az.network/stop-azapplicationgateway?view=azps-6.4.0

it does not delete the settings and is much faster to start vs provision all anew

**Comment by marrobi** [comment]
Ah, didn't know could do that - so do you want commands like:

`make stop-tre`, `make start-tre`?

This would stop/start app gateway, firewall, VMs, anything else?

**Comment by lenisha** [comment]
yes that would be great to have.

and could ServiceBus (which cant be stopped) be replaced with EventHubs Standard SKU much cheaper and has private endpoints ?

**Comment by marrobi** [comment]
Hmm, I can't see a way to do it with the `az` cli, can you see one? If not will have to add PowerShell to the dev container.

Maybe you could put a script together and I could integrate it into the make file?

**Comment by marrobi** [comment]
@lenisha want to give this a try: https://github.com/microsoft/AzureTRE/pull/973

Once this is merged can add further operations - say stop VMs to the script.

**Comment by marrobi** [comment]
I'm going to split this into three stories:

1. As a developer I want to stop charges for as many core resources as possible at the end of each day.
2. As a TRE Administrator I want to disable a workspace, and all child resources also get disabled. 
3. As a developer I want to save money by deploying less secure product SKUs with the understanding that this means that I am compromising on security

**Comment by tamirkamara** [comment]
The bulk of the work is done and other stuff have their own issue. So will close this.
